### PR TITLE
[WIP] Shadowlings, except this time it's more balanced

### PR DESCRIPTION
**Pull request by ghost**
:cl: More Robust Than You
add: Shadowlings!
/:cl:

no description since this is not in a final state yet

### CODE REVIEW

**Comment by Partheo** [comment]
PR descriptions like this should be a crime

**Comment by Xhuis** [comment]
Trying to reintroduce a gamemode almost universally hated for awful balance by claiming it's now balanced, then not explaining why in a detailed way that makes sense, is a little shortsighted, don't you think?

This is the third time people have tried to bring back shadowlings, and the other two times had a lot more going for them. We already have nightmares to build off of. Just let it go! Shadowling doesn't need to come back!

**Comment by epochayur** [comment]
not reading 100% of your code to try to figure out what you changed

File: code/game/gamemodes/shadowling/shadow_abilities.dm [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+		target.visible_message("<span class='danger'>[target] suddendly collapses...</span>")
+		to_chat(target, "<span class='userdanger'>A purple light flashes across your vision, and you lose control of your movements!</span>")
+		target.Stun(100)
+		M.silent += 10
+	else //Distant glare
+		var/loss = 100 - ((distance - 1) * 18)
+		target.adjustStaminaLoss(loss)
+		target.stuttering = loss
+		to_chat(target, "<span class='userdanger'>A purple light flashes across your vision, and exhaustion floods your body...</span>")
+		target.visible_message("<span class='danger'>[target] looks very tired...</span>")
```
Comment (Xhuis) [inline code/game/gamemodes/shadowling/shadow_abilities.dm]: This is a lot better than what Glare used to be. That said, stamina loss is inelegant because of how weird it is (pseudo-health) and the power curve over distance here is really weird, with mostly insignificant losses until about 4 tiles away, at which point the ability doesn't become worth using at all due to how ineffectual it is. `loss` also isn't properly clamped, meaning that this could cause targets to *heal* stamina damage.

**Comment by MoreRobustThanYou** [comment]
tl;dr: removed life drain, glare now uses stamina damage, etc etc

File: code/game/gamemodes/shadowling/shadow_abilities.dm [hunk 2]
```diff
@@ -0,0 +1,542 @@
...
+			for(var/obj/item/F in H)
+				F.disable_lights()
+		for(var/obj/machinery/camera/cam in T.contents)
+			cam.set_light(0)
+			if(prob(10))
+				cam.emp_act(2)
+		for(var/mob/living/silicon/robot/borg in T.contents)
+			if(!borg.lamp_cooldown)
+				borg.update_headlamp(TRUE, INFINITY)
+				to_chat(borg, "<span class='danger'>Your headlamp is fried! You'll need a human to help replace it.</span>")
```
Comment (Xhuis) [inline code/game/gamemodes/shadowling/shadow_abilities.dm]: I think this is fair for nightmares because nightmares have to land a hit. For shadowlings, this effectively makes borgs completely useless against shadowlings due to how low of a cooldown Veil has.

File: code/game/gamemodes/shadowling/shadow_abilities.dm [hunk 3]
```diff
@@ -0,0 +1,542 @@
...
+	for(var/obj/structure/glowshroom/G in view(7, user)) //High radius because glowshroom spam wrecks shadowlings
+		G.visible_message("<span class='warning'>[G] withers away!</span>")
+		qdel(G)
+
+////////////////////
+////FLASH FREEZE////
+////////////////////
+/obj/effect/proc_holder/spell/flashfreeze //Stuns and freezes nearby people - a bit more effective than a changeling's cryosting. but doesn't last nearly as long
+	name = "Flash Freeze"
+	desc = "Instantly freezes the blood of nearby people, stunning them for a short moment and causing burn damage."
```
Comment (Xhuis) [inline code/game/gamemodes/shadowling/shadow_abilities.dm]: This ability is one of the reasons old shadowlings were terrible. This is one of a ton of escape abilities and this one is formidably strong, not only causing items to drop but also to massively freeze people nearby.

File: code/game/gamemodes/shadowling/shadow_abilities.dm [hunk 4]
```diff
@@ -0,0 +1,542 @@
...
+
+
+
+//////////////////////
+////SLING HIVEMIND////
+//////////////////////
+
+/obj/effect/proc_holder/spell/shadowling_hivemind //Lets a shadowling talk to its allies
+	name = "Hivemind Commune"
+	desc = "Allows you to silently communicate with all other shadowlings and thralls."
```
Comment (Xhuis) [inline code/game/gamemodes/shadowling/shadow_abilities.dm]: This is probably best refactored using language systems so you can put a `.b` or something before your message and it'll send to the Hivemind. I was able to do it for umbrages when I was working on them, so it's absolutely doable.

File: code/game/gamemodes/shadowling/shadow_abilities.dm [hunk 5]
```diff
@@ -0,0 +1,542 @@
...
+
+
+
+///////////////////
+////BLIND SMOKE////
+///////////////////
+
+/obj/effect/proc_holder/spell/blindness_smoke //Spawns a cloud of smoke that blinds non-thralls/shadows and grants slight healing to shadowlings and their allies
+	name = "Blindness Smoke"
+	desc = "Spews a cloud of smoke which will blind enemies."
```
Comment (Xhuis) [inline code/game/gamemodes/shadowling/shadow_abilities.dm]: See Flash Freeze. This is an escape ability, and going overboard will only make history repeat.

File: code/game/gamemodes/shadowling/shadow_abilities.dm [hunk 6]
```diff
@@ -0,0 +1,542 @@
...
+
+
+
+/////////////////////
+////SONIC SCREECH////
+/////////////////////
+
+/obj/effect/proc_holder/spell/unearthly_screech //Damages nearby windows, confuses nearby carbons, and outright stuns silly cones
+	name = "Sonic Screech"
+	desc = "Deafens, stuns, and confuses nearby people. Also shatters windows."
```
Comment (Xhuis) [inline code/game/gamemodes/shadowling/shadow_abilities.dm]: This ability was meant to counter cyborgs. Since headlamp destruction is now built into Veil, I recommend choosing one or the other; with both cyborgs will become worse than useless - they'll become dead weight.

File: code/game/gamemodes/shadowling/shadow_abilities.dm [hunk 7]
```diff
@@ -0,0 +1,542 @@
...
+			W.take_damage(rand(80, 100))
+
+
+/////////////////////
+////SHUTTLE DELAY////
+/////////////////////
+
+/obj/effect/proc_holder/spell/targeted/shadowling_extend_shuttle
+	name = "Destroy Engines"
+	desc = "Extends the time of the emergency shuttle's arrival by fifteen minutes. This can only be used once."
```
Comment (Xhuis) [inline code/game/gamemodes/shadowling/shadow_abilities.dm]: This was a horrible terrible ability that served as a band-aid fix to the larger problem of shuttle calls. Shuttle calls were the only way to combat the snowballing that shadowlings had with thralls, and due to shadowling's nature it still is. Nightmares circumvented this because the only way to put out lights is to hit someone with the light eater, making it more viable to attack with lights, but due to shadowlings having so many escape abilities they're very difficult to actually kill or pin down. 

Making it more feasible to actually hunt down the shadowlings, as well as making deconversion less tedious, is a far preferable solution; this is a badly-made ability that only serves to frustrate everyone involved: shadowlings are frustrated because they only put off the shuttle, and crew are frustrated because they have no way to fight back against it.

**Comment by Repukan** [comment]
Still another binary conversion mode.

File: code/game/gamemodes/shadowling/thrall_abilities.dm [hunk 8]
```diff
@@ -0,0 +1,121 @@
+
+/obj/effect/proc_holder/spell/targeted/sling/lesser_glare //a defensive ability, nothing else. can't be used to stun people, steal tasers, etc. Just good for escaping
+	name = "Lesser Glare"
+	desc = "Makes a single target dizzy for a bit."
```
Comment (Xhuis) [inline code/game/gamemodes/shadowling/thrall_abilities.dm]: Because of the snowballing nature of shadowlings, removing this ability's stunning capability still doesn't fix it. Taunting Tirade was removed from CC because it was a guaranteed escape, and Tirade had drawbacks and could be stopped with a disarm. This one doesn't and can't. 25 confusedness is a massive amount, and likely enough to keep the target confused well after the ability is ready again.

File: code/game/gamemodes/shadowling/thrall_abilities.dm [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+	else
+		revert_cast()
+
+////////////////////
+////THRALL GUISE////
+////////////////////
+
+/obj/effect/proc_holder/spell/self/lesser_shadow_walk //Thrall version of Shadow Walk, only works in darkness, doesn't grant phasing, but gives near-invisibility
+	name = "Guise"
+	desc = "Wraps your form in shadows, making you harder to see."
```
Comment (Xhuis) [inline code/game/gamemodes/shadowling/thrall_abilities.dm]: This ability was kind of useless, especially on a massive two-minute cooldown. I don't know what to replace it with, though.

File: code/modules/mob/living/carbon/human/species_types/shadowpeople.dm [hunk 10]
```diff
@@ -68,6 +73,41 @@
 /datum/species/shadow/nightmare/check_roundstart_eligible()
 	return FALSE
 
+/datum/species/shadow/ling
+	//Normal shadowpeople but with enhanced effects
+	name = "Shadowling"
+	id = "shadowling"
```
Comment (Xhuis) [inline code/modules/mob/living/carbon/human/species_types/shadowpeople.dm]: Make sure to blacklist this from mutations/magic mirrors/etc.

**Review by ghost — COMMENTED** [review]
As the original maker of shadowling and its failed reboot, here are my problems with this PR right now.

**Comment by Xhuis** [comment]
Ultimately, I seriously just don't think shadowling can work. It's fundamentally flawed because of how much it snowballs, and the original concept (stealth, subterfuge, and paranoia) don't really work because they have to maintenance crawl for more or less the entire shift due to 360-degree vision. Thralls and masks were a good way, but Lesser Glare ruined that, and in its current state it still will even without its stunning.

File: code/game/gamemodes/shadowling/shadow_abilities.dm (hunk 4 above)
Comment (MoreRobustThanYou) [reply code/game/gamemodes/shadowling/shadow_abilities.dm]: Saymode datums?

**Comment by MoreRobustThanYou** [comment]
good point @Xhuis
